src/linux/py_thread.h

```diff
@@ -50,9 +50,9 @@ _infer_tid_field_offset(py_thread_t * py_thread) {
       PTHREAD_BUFFER_SIZE * sizeof(void *),
       _pthread_buffer
   ))) {
-    for (register int i = 0; i < PTHREAD_BUFFER_SIZE; i++) {
+    for (register int i = 0; i < PTHREAD_BUFFER_SIZE * sizeof(void *) / sizeof(pid_t); i++) {
       log_d("pthread_t at %p", py_thread->tid);
-      if (py_thread->raddr.pid == (uintptr_t) _pthread_buffer[i]) {
+      if (py_thread->raddr.pid == *((pid_t *) _pthread_buffer + i)) {
```

Comment:
I think if we use that as a fallback, that does not really protect you from accidentally matching something else, as it if the comparing uintptr_t fails, it would fallback to the comparing pid_t logic..

Do you think if it good we parse the output of gnu_get_libc_version(), as we can surely know from which glibc version it started to replace pid with pid_unused. So for glibc versions above that we compare uintptr_t, for lower glibc versions we compare pid_t.

Reply:
> Do you think if it good we parse the output of gnu_get_libc_version()

This sounds good to me. Can we make it so that it is also compatible with musl as I'd like to start making these builds available too from releases? We should be able to import `features.h` with both glibc and musl, and check for [`__GLIBC__` and `__GLIBC_MINOR__`][macros] (which _en passant_ might save us from having to parse the string returned by `gnu_get_libc_version`?).

[macros]: https://sourceforge.net/p/predef/wiki/Libraries/

Reply:
hmm I think the macro resolves to a fixed version at compile time, so the built binary is not portable. but the function could work at runtime. I will need to furthur check though.

Reply:
> hmm I think the macro resolves to a fixed version at compile time

Ah, of course! We're not linking libc statically so indeed we'd have to call the function. But at least we should be able to use the macros to determine what we're compiling against 🤞.